docs/user/openstack/affinity.md

```diff
@@ -0,0 +1,136 @@
+```
+
+This command will interactively prompt for the required information, if an
+`install-config.yaml` file is not present in the current working directory.
+
+### Step 3: Edit the MachineSet manifests
+
+The "create manifests" command creates several files, among which the worker
+MachineSet definition in
+`openshift/99_openshift-cluster-api_worker-machineset-0.yaml`.
+
+We must add a new property to the MachineSet: `serverGroupID`. That property lives under `spec.template.spec.providerSpec.value`.
+
+**Use the Server group id from step 1.**
+
+I have used the example UUID `aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee` in the template below.
+
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+    machine.openshift.io/cluster-api-machine-role: <node_role>
+    machine.openshift.io/cluster-api-machine-type: <node_role>
+  name: <infrastructure_ID>-<node_role>
+  namespace: openshift-machine-api
+spec:
+  replicas: <number_of_replicas>
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+        machine.openshift.io/cluster-api-machine-role: <node_role>
+        machine.openshift.io/cluster-api-machine-type: <node_role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+    spec:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          cloudsSecret:
+            name: openstack-cloud-credentials
+            namespace: openshift-machine-api
+          flavor: <nova_flavor>
+          image: <glance_image_name_or_location>
+          serverGroupID: aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee
+          kind: OpenstackProviderSpec
+          networks:
+          - filter: {}
+            subnets:
+            - filter:
+                name: <subnet_name>
+                tags: openshiftClusterID=<infrastructure_ID>
+          securityGroups:
+          - filter: {}
+            name: <infrastructure_ID>-<node_role>
+          serverMetadata:
+            Name: <infrastructure_ID>-<node_role>
+            openshiftClusterID: <infrastructure_ID>
+          tags:
+          - openshiftClusterID=<infrastructure_ID>
+          trunk: true
+          userDataSecret:
+            name: <node_role>-user-data
+          availabilityZone: <optional_openstack_availability_zone>
+```
+
+### Step 4: Complete the installation
+
+After saving the MachineSet file in place, use `openshift-install` to complete
+the installation.
+
+Note: running the following command has the side effect of deleting both
+`openshift` and `manifests` directories from the current working directory.
```

Comment:
Aside: just the fact that we need to point this out to reduce user surprise is a massive smell 🙄

Reply:
@pierreprinetti perhaps you already know about this but your modifications are not deleted, they're saved under another form in the `.openshift_install_state.json` file. You can re-generate the `openshift` and `manifests` directories re-running the `openshift-install create manifests` command if you like.

Same thing with your `install-config.yaml`, you can retrieve it re-running the `openshift-install create install-config` command.

Reply:
WAT